src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -2,34 +2,123 @@
+ * color and rotations)
  */
 public class Elevator extends SubsystemBase {
 
     private final CANSparkMax elevatorMotor = new CANSparkMax(Const.CAN.ELEVATOR_MOTOR, MotorType.kBrushless);
     private final WPI_VictorSPX wheelMotor = new WPI_VictorSPX(Const.CAN.ELEVATOR_WHEEL);
     private final ColorSensorV3 colorSensor = new ColorSensorV3(I2C.Port.kOnboard);
+    private final CANPIDController elevatorPID = new CANPIDController(elevatorMotor);
+    private String targetColor = "";
+    private int revCount = 0;
+    private boolean done = false;
+
 
     public Elevator() {
 
     }
+
+    /**
+     * Position loop control for the elevator.
+     * 
+     * @param position Encoder reference for elevator setpoint.
+     */
+    public void setElevatorPosition(double position) {
+        elevatorPID.setP(Const.PID.ELEVATOR_P);
+        elevatorPID.setI(Const.PID.ELEVATOR_I);
+        elevatorPID.setReference(position, ControlType.kPosition);
+    }
+
+    /**
+     * Velocity loop control for the elevator.
+     * 
+     * @param velocity Speed reference for elevator.
+     */
+    public void setElevatorVelocity(double velocity) {
+        elevatorPID.setP(Const.PID.ELEVATOR_P);
+        elevatorPID.setI(Const.PID.ELEVATOR_I);
+        elevatorPID.setReference(velocity, ControlType.kVelocity);
+    }
+
+    /**
+     * Sets the color wheels speed.
+     * 
+     * @param speed The speed, between -1 and 1, of the motor.
+     */
+    public void setWheelSpeed(double speed) {
+        wheelMotor.set(speed);
+    }
+
+    /**
+     * Spins color wheel to meet color and revolution needs.
+     * 
+     * @param targetRevs Number of revolutions desired.
+     * @param colorMode  Set true if we need to match the color on the wheel.
+     */
+    public void spinColorWheel(int targetRevs, boolean colorMode) {
+        /** Counts revs with every color change. */
+        String color = colorSensor.getColor().toString();
+        if (colorSensor.getColor().toString() != color) {
+            revCount++;
+            color = colorSensor.getColor().toString();
+        }
+        /** Only counting revolutions, no color. */
+        if (!colorMode) {
+            while (revCount < targetRevs * 8) {
+                wheelMotor.set(Const.Speed.COLOR_WHEEL_FIXED_SPEED);
+            }
+        /** Counting revolutions with the color. */
+        } else if (colorMode) {
+            if (color != colorToTargetColor() && !done) {
+                wheelMotor.set(Const.Speed.COLOR_WHEEL_FIXED_SPEED);
+            } else {
+                done = true;
+                while (revCount < targetRevs * 8) {
+                    wheelMotor.set(Const.Speed.COLOR_WHEEL_FIXED_SPEED);
+                }
+            }
+        }
+    }
+
+    /**
+     * Serves only the purpose of finding the offset targeted color for the function
+     * spinColorWheel().
+     */
+    public String colorToTargetColor() {
```

Comment:
This should look something like `private WheelColor offsetColor(WheelColor color)`, where the target color is returned based on the provided color. This will provide us with more flexibility.

Reply:
But the provided color doesn't necessarily need to be a parameter since it's going to be retrieved from the same place anyways, the game information. I see that you had a comment about that as well, that it shouldn't be done in a function, but it seems to me that it's just skipping a step that we will have to take eventually.

Reply:
But it does: the `spinWheel()` function that will take a color object will pass that object into the `offsetColor()` function to get the color offset. We will pass in the color via a command, and this will be responsible for calculating the offset based on what the command provides. Remember that subsystems are just interfaces to commands. They don't gather information from outside components, that's the command's job.

Reply:
I get what you mean, instead of reaching outside in functions we need that outside information as parameters. Gotcha.